packages/pluggableWidgets/tree-view-web/src/components/TreeView.tsx

```diff
@@ -0,0 +1,100 @@
+import { GUID } from "mendix";
+import { createElement, CSSProperties, HTMLAttributes, ReactElement, ReactNode, useState } from "react";
+import classNames from "classnames";
+
+import "../ui/TreeView.scss";
+
+interface TreeViewObject {
+    id: GUID;
+    value: string | undefined;
+    content: ReactNode;
+}
+
+export interface TreeViewProps {
+    class: string;
+    style?: CSSProperties;
+    items: TreeViewObject[];
+    isUserDefinedLeafNode: TreeViewBranchProps["isUserDefinedLeafNode"];
+    startExpanded: TreeViewBranchProps["startExpanded"];
+}
+
+export function TreeView({
+    class: className,
+    items,
+    style,
+    isUserDefinedLeafNode,
+    startExpanded
+}: TreeViewProps): ReactElement {
+    // TODO: for lazy loading/knowing whether there are children, it might be better to not render any DOM here if there are no items.
+    return (
+        <div className={classNames("mx-tree-view", className)} style={style}>
+            {items.map(({ id, value, content }) => (
+                <TreeViewBranch
+                    key={id}
+                    value={value}
+                    isUserDefinedLeafNode={isUserDefinedLeafNode}
+                    startExpanded={startExpanded}
+                >
+                    {content}
+                </TreeViewBranch>
+            ))}
+        </div>
+    );
+}
+
+interface TreeViewBranchProps {
+    isUserDefinedLeafNode: boolean;
+    startExpanded: boolean;
+    value: TreeViewObject["value"];
+    children: TreeViewObject["content"];
+}
+
+function getTreeViewHeaderAccessibilityProps(isLeafNode: boolean): HTMLAttributes<HTMLHeadingElement> {
+    if (isLeafNode) {
+        return {};
+    }
+    return {
+        tabIndex: 0,
+        role: "button"
+    };
+}
+
+function TreeViewBranch(props: TreeViewBranchProps): ReactElement {
+    const [treeViewIsExpanded, setTreeViewIsExpanded] = useState<boolean>(props.startExpanded);
+
+    function toggleTreeViewContent(): void {
+        if (!props.isUserDefinedLeafNode) {
+            setTreeViewIsExpanded(isExpanded => !isExpanded);
+        }
+    }
+
+    const headerAccessibilityProps = getTreeViewHeaderAccessibilityProps(props.isUserDefinedLeafNode);
+
+    return (
+        <div className="tree-view-branch">
```

Comment:
The convention is to start classes with `widget-`

Reply:
I thought we only did that for the top level class name, not for everything passed that. Am I mistaking?

Reply:
No, we use the prefix `widget-` for all new widgets, like accordion. When we convert widgets, we stick as much as possible to the previous class names. Many of these converted widgets include the same prefix, like progress-circle, progress-bar and switch. So I would apply the prefix everywhere :)